File: api/internal/domain/account/organization.go
package account

import "nathan.dev/consent/internal/domain"

type Organization struct {
	Name    string
	Members map[string]*OrganizationMember
}

type OrganizationModel struct {
	domain.Model
	Organization
}

type OrganizationMember struct {
	Role Role
}

type Role struct {
	slug string
}

func (r Role) String() string {
	return r.slug
}

var (
	Owner = Role{"owner"}
)

File: src/Consent.Domain/Accounts/Organization.cs
namespace Consent.Domain.Accounts
{
    public record Organization
    {
        public string Name { get; init; }

        public Organization(string name)
        {
            Name = name;
        }
    }

    public record OrganizationEntity : Organization, IEntity
    {
        public int Id { get; init; }

        public OrganizationEntity(int id, Organization organization) : base(organization)
        {
            Id = id;
        }
    }
}
